lib/inspec/utils/filter.rb

```diff
@@ -256,7 +256,7 @@ def matches_regex(x, y)
     end
 
     def matches(x, y)
-      x === y # rubocop:disable Style/CaseEquality
+      y === x # rubocop:disable Style/CaseEquality
```

Comment:
Would you mind explaining the fix?

Reply:
Yes sure. So this method is called when the filter table needs to match anything other than an Integer, float & regex. So this is the `else` condition. And since this one was called in case of **ranges**. It was breaking because I found it was checking initially for example `1 === 1..2` while my change checks for `1..2 === 1` and this returns true because this is how it should be used for this case. I referred [this](https://stackoverflow.com/questions/4467538/what-does-the-operator-do-in-ruby).